packages/asc-ui/src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -5,24 +5,43 @@ import MenuList from './ContextMenuList'
 import { KeyboardKeys } from '../../types'
 import ownerDocument from '../../utils/ownerDocument'
 
+const selectedChildInitial = -1
+
 type Props = {
   position?: AscCore.ContextMenuTypes.Position
   label?: string
   icon?: React.ReactNode
+  open?: boolean
 }
 
-type State = {}
-
-const selectedChildInitial = -1
+type State = {
+  open?: boolean
+  selectedChild: number
+}
 
 class ContextMenu extends React.Component<Props, State> {
-  state = {
+  static defaultProps = {
     open: false,
-    selectedChild: selectedChildInitial,
   }
 
-  list = React.createRef<HTMLDivElement>()
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      open: props.open,
+      selectedChild: selectedChildInitial,
+    }
+  }
 
+  componentDidUpdate(prevProps: Props) {
+    const { open } = this.props;
+
+    if (prevProps.open !== open) {
+      this.setState({ open })
+    }
+  }
+
+  list = React.createRef<HTMLDivElement>()
```

Comment:
Member name is unclear, for list and root, what are they for? 
Please move the member declarations to the begining of the class definition

Reply:
LInting warning: Enforce component methods order (react/sort-comp)

Reply:
Agree, replaced `root` with `wrapper`. So `wrapper` is for `ContextMenuWrapper` and `list` for `ContextMenuList`